src/manpages/airoscwordlist.1
.TH AIROSCWORDLIST 1 "May 5, 2008"

.SH NAME
airoscwordlist - Create wordlists from mac.

.SH DESCRIPTION
airoscwordlist reads a list of macs with an associated command
and creates the wordlist according to the inputted mac.

.SH SYNOPSIS
.B airoscwordlist
[options]

.SH OPTIONS
.TP
	--mac-address|-m
	[--essid|-e] 
	[--filename|-f]
	[--company|-c]
	--companyfile|-cf
	[--verbose|-v]
	[--version|-V]

.SH AUTHOR
Airoscwordlist was written by David Francos Cuartero (XayOn) <yo.orco@gmail.com>
.br
This manual page was written by David Francos Cuartero <yo.orco@gmail.com>.

.SH SEE ALSO
.br
.B airdecap-ng(1)
.br
.B airdriver-ng(1)
.br
.B aireplay-ng(1)
.br
.B airmon-ng(1)
.br
.B airodump-ng(1)
.br
.B airolib-ng(1)
.br
.B airsev-ng(1)
.br
.B airtun-ng(1)
.br
.B buddy-ng(1)
.br
.B easside-ng(1)
.br
.B ivstools(1)
.br
.B kstats(1)
.br
.B makeivs-ng(1)
.br
.B packetforge-ng(1)
.br
.B wesside-ng(1)
.br
.B aircrack-ng(1)



CHANGELOG
Changelog for 2.0.9
-fixed client and clientless attacks chopchop and frag packet injection not generating IVs because -h MAC was not set
-force frag, chopchop and aircrack-ng xterms to stay on screen after successfull command so you can see results
-cleaned the code some more
-cleaning functions needs more tweaks
-atheros users need to use options 9 (monitor mode) before other commands to create-destroys correct VAPS
-fixed "unary operator expected" msg (harmless bug) (not fully fixed because of "" generating errors in logic tests)
-tweaked 2 steps attacks (frag, chopchop) to go to menu to inject after successfull first step
-added option to jump to injection menu (added this to the main menu)
-fixed AP encryption detection for cracking step (bug was introduced by "")
-lowered adapter speed to 1M for wesside-ng attacks
-added -hold to mdk attack to allow users to see results before the windows closes

Changelog for 2.0.8 (27/01/2008)
-added wesside-ng airoscript style usage
-added interface type detection
-tweaked menu
-first attempt at making airoscript work with different adapters (add more card support)(BIG THX TO GREEN-FREQ!!!)
-atheros device are now supported
-added mdk3
-added back to main menu options
-fixed chopchop attack (BIG THX TO GREEN-FREQ!!!)
-tweaked FAKE_MAC function for atheros

Changelog for 2.0.7 BETA1 (21/01/2007)
-changed capture from IVs to full date (ivs to cap)
-updated script for new implementation of fragmentation attack
-other minor modifications and tweaks

Changelog for PRE-1.8.2-BETA1 (30/11/2006)
-fixed all commands leading to : "head: Binary: invalid number of lines" (MAJOR THX TO ANDREA)
-fixed all commands where SSID was truncated (MAJOR THX TO PILOTSNIPES)

Changelog for 1.8.1 (03/11/2006)
-bug fix release
-fixed some problem in the SSID detection function (MAJOR THX TO andreahaku)
-cleaned a few more typos and bugs
-documentation completed a bit

Changelog for 1.8 (17/10/2006)
-dropped support for aircrack-ng 0.6.1 and older
-will only work with patched version of aircrack-ng 0.6.2 (patches can be found in the forum under the programming section)
-specifique channels hopping (for example channel 1,6,11 or 1-3,5-7,9,11)
-Access point filtering during attack (airodump will only list attacked Access point)
-new SSID detection, works with ALL ssid (supports all character and spacing possible)
-new Access point selection menu that list all AP with encryption type (eliminates the WEP? problem)
-option 2) now get you directly to a full list of AP
-chopchop attack and fragmentation attack implement
-WPA dictionnary attack implemented
-changed menu 3,4,5 based on detection of AP encryption
-added option to jump to associated client list from ap selection
-color change support for xterm
-Windows size and placement fully changeable from variables settings
-changed filenames from SSID to MAC because of possible issues if weird SSID (to eliminate security menace (if SSID is something like "rm -rf /*" etc...)
-added option an option to run "1) scan 2) select 3) attack" one after the other
-added some comments in the script to help users
-added greetings message (can be disabled setting WELCOME="1" to "0")

Changelog for 1.7RC7 (24/09/2006)
-added some resilience to the configure function now works properly with hidden ssid's and rausb devices 

Changelog for 1.7RC6 (19/09/2006)
-Code cleanup
-added deauth command to solointeractiveattack function
-changed base variables (fudge factor & inject rate)

Changelog for 1.7RC5 (14/09/2006)
-Fixed minor cosmetic issue caused by carriage return char when ssid is missing (15/9/06)
-Fixed the weird character match issue!! edit at will!!!
-added an abitrary variable ($acv) to the aircrackversioncheck function which can be used in future to keep the script backwards compatible (had to guess at what version 0.6.0 of aircrack-ng displays when run on it's own? and version 0.5.so might not work??)
-used new variable in crack function to check version and run correct crack command for the aricrack version installed fixing the bug on the B|T forum.

Changelog for 1.7RC4 (13/09/2006)
-corrected minor speellling issues ;o) 
-added aircrack-ng version detection function currently all it does is display aircrack version in the menu function (should be useful when new functionality is added to aircrack to allow backwards compatibility)
-added solointeractiveattack function (designed to run just the airreplay interactive attack function only for use with an attack that has stalled i.e. data has stopped increasing, the user can try this for another injection thread! (might not actually work?? needs more testing does spawn and seems to work!)
-added option to manually add ssid if a blank ssid is detected (this has some problems when certain editors are used as this is not strictly blank but a non printable character recommend kwrite on Back|track to edit script)


src/tools/airoscwordlist.pl
#!/usr/bin/perl
# Airoscwordlist.pl - Wordlist generator for airoscript

=pod

  Copyright (C) 2009 David Francos Cuartero
        This program is free software; you can redistribute it and/or
        modify it under the terms of the GNU General Public License
        as published by the Free Software Foundation; either version 2
        of the License, or (at your option) any later version.

        This program is distributed in the hope that it will be useful,
        but WITHOUT ANY WARRANTY; without even the implied warranty of
        MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
        GNU General Public License for more details.

        You should have received a copy of the GNU General Public License
        along with this program; if not, write to the Free Software
        Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA. at airoscwordlist.pl line 31.

=cut

use strict;use warnings;use Getopt::Long;use YAML::Syck qw(LoadFile);

GetOptions("mac-address|m:s"=>\(my $mac),"ssid|s:s"=>\(my $essid),
"filename|f:s" => \(my $filename),"company|c:s" => \(my $company),
"companyfile|cf:s" => \(my $cf),"verbose" => \(my $verbose),
"version|V" => \(my $getversion),"license|L" =>\(my $license));

my $executable;

	# Main Info here.
my $usage="$0 [OPTIONS]\nOPTIONS:\n\t--mac-address|-m\n\t".
"[--essid|-e] \n\t[--filename|-f]\n\t[--company|-c]\n\t--companyfile|-cf\n\t".
"[--verbose|-v]\n\t[--license|-L]\n\t[--version|-V]\n";my $version="1.0";

my $licensewarn="Copyright (C) 2009 David Francos Cuartero
\tThis program is free software; you can redistribute it and/or
\tmodify it under the terms of the GNU General Public License
\tas published by the Free Software Foundation; either version 2
\tof the License, or (at your option) any later version.";
	my $extendedlw="\n\n\tThis program is distributed in the hope that it will be useful,
\tbut WITHOUT ANY WARRANTY; without even the implied warranty of
\tMERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
\tGNU General Public License for more details.
\n\tYou should have received a copy of the GNU General Public License
\talong with this program; if not, write to the Free Software
\tFoundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.";

die "Airoscwordlist ".$version."\n".$licensewarn if $getversion;
die $licensewarn.$extendedlw if $license;
die $usage if !$mac or !$cf;

my $comp=LoadFile($cf) or die "ERROR, could not open company file: $!";

sub _guess_company(){
	my $c_mac;for(keys %$comp){$c_mac=$_ if $mac=~/$_/;}
	if ($comp->{$c_mac}){
		return $comp->{$c_mac}{'name'},$comp->{$c_mac}{'exec'};
	} else { exit(404); }
}

sub _execute_wholething(){my $_=shift;
	s/FILENAME/$filename/ if $filename;
	s/MACADDR/$mac/ if $mac;
	s/ESSID/$essid/ if $essid;
	print;
	system($_);
}


$filename=$ENV{'HOME'}."/generated-wordlist-".localtime(time) if !$filename;
$cf="/usr/share/airoscript/default_company_names" if !$cf;

($company,$executable)=&_guess_company($mac) if !$company;
print "Company recognised: $company, executable $executable" if $verbose;

&_execute_wholething($executable);

__DATA__

=head1 NAME

  Airoscwordlist - Automatic wordlist generator for airoscript.

=head1 USAGE

  Airoscwordlist [ OPTIONS ]

=head1 OPTIONS

        --mac-address|-m	Target mac address
        [--essid|-e] 		Target essid
        [--filename|-f]		Filename to write wordlist
        [--company|-c]		Company to look for
        --companyfile|-cf	Company file
        [--verbose|-v]		Enable verbosity
        [--license|-L]		Show license and exit
        [--version|-V]		Show version and exit

=head1 COMPANY FILE

  Company file is just a yaml file with the format:

  "00:00:00": # Those are the last 6 digits of mac addr
  		name: CompanyName
		exec: CommandToExecute ESSID FILENAME MAC 
  
  Where ESSID, FILENAME and MAC are replaced by options-specified ones.

=head1 FUNCTIONS

=over

=item _guess_company()

	Processes yaml file and returns exec and name from company.

=item _execute_wholething()

  Replaces MAC, ESSID and FILENAME by options-specified ones.
  and executes command given by _guess_company

=back
